### PR TITLE
feat(sponsor): implement sponsor list on index page

### PR DESCRIPTION
**Pull request by mattwang44**
## Types of Changes

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description

- Add mock data for sponsor list API.
- Implement sponsor list on index page & affiliated modal.

### Preview: Sponsor List

<details>
<summary>web view</summary>

</details>

<details>
<summary>mobile view (iPhone 6/7/8)</summary>

</details>

### Preview: Modal

<details>
<summary>web view</summary>

</details>

<details>
<summary>mobile view (iPhone 6/7/8)</summary>

</details>

### CODE REVIEW

File: components/sponsors/SponsorModal.vue [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        padding: 60px 48px 48px 60px;
+    }
+    height: 80%;
+    background-color: #121023;
+}
+
+.lightBox__closeButton {
+    @apply relative flex justify-end text-yellow-500;
+    top: 10px;
+    right: 10px;
```
Comment (flynn-hou) [inline components/sponsors/SponsorModal.vue:77]:
```suggestion
    top: -40px;
    right: -20px;
```

This looks closer to the design spec, give or take.

File: components/sponsors/SponsorModal.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+.lightBox__title {
+    @apply flex justify-start items-center;
```
Comment (flynn-hou) [inline components/sponsors/SponsorModal.vue:81]: Remove the unnecessary `justify-start` class as the `justify-content` rule by default is already `flex-start`.

File: pages/index.vue [hunk 3]
```diff
@@ -26,56 +26,81 @@
...
+    },
+    methods: {
+        showModal(sponsor) {
+            this.isOpened = true
+            this.selectedSponsor = sponsor
+        },
+        closeModal() {
+            this.isOpened = false
+            this.selectedSponsor = {}
+        },
```
Comment (flynn-hou) [inline pages/index.vue:97]: Remove this unused method

File: components/sponsors/SponsorCard.vue [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<template>
+    <div v-show="!!logoUrl" :class="classObject">
+        <img :src="logoUrl" class="object-contain w-full" />
```
Comment (flynn-hou) [inline components/sponsors/SponsorCard.vue:3]:
```suggestion
        <img :src="logoUrl" alt="" class="object-contain w-full" />
```

Provide the required attribute for ``

File: components/sponsors/SponsorModal.vue [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                <sponsor-card :logo-url="context.logo_url" small></sponsor-card>
+                <h1 class="text-xl mx-8 border-b-8 border-solid">
+                    {{ context.name }}
+                </h1>
+            </div>
+            <h2 class="text-xl text-yellow-500 mt-4">{{ $t('about') }}</h2>
+            <div class="lightBox__intro">
+                <p class="text-sm">{{ getAttributeByLocale('intro') }}</p>
+            </div>
+            <div class="lightBox__bottons">
```
Comment (flynn-hou) [inline components/sponsors/SponsorModal.vue:15]:
```suggestion
            <div class="lightBox__buttons">
```

Fix typo

**Review by flynnhou — APPROVED** [review]
lgtm

**Comment by mattwang44** [comment]
[Implement sponsor list](https://trello.com/c/LSEujtqr/47-implement-sponsor-list)

**Pull request merged**